### PR TITLE
Use .entrySet() instead of keySet()

### PR DESCRIPTION
**Pull request by asashour**

### CODE REVIEW

File: src/edu/stanford/nlp/loglinear/model/ConcatVectorNamespace.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Map.Entry;
```
Comment (AngledLuffa) [inline src/edu/stanford/nlp/loglinear/model/ConcatVectorNamespace.java]: I don't like this.  Map.Entry in the code is fine

File: itest/src/edu/stanford/nlp/dcoref/DcorefExactOutputITest.java [hunk 2]
```diff
@@ -106,11 +106,10 @@ private static void saveResults(String filename, Map<Integer, CorefChain> chains
     FileWriter fout = new FileWriter(filename);
     BufferedWriter bout = new BufferedWriter(fout);
 
-    List<Integer> keys = new ArrayList<>(chains.keySet());
-    Collections.sort(keys);
+    Map<Integer, CorefChain> sorted = new TreeMap<>(chains);
```
Comment (AngledLuffa) [inline itest/src/edu/stanford/nlp/dcoref/DcorefExactOutputITest.java]: I don't expect this to be as optimized as to Collections.sort.  Seems like an unnecessary change

Comment (asashour) [reply itest/src/edu/stanford/nlp/dcoref/DcorefExactOutputITest.java]: It's a test, anyhow reverted.

File: src/edu/stanford/nlp/optimization/GoldenSectionLineSearch.java [hunk 3]
```diff
@@ -161,10 +161,8 @@ public double minimize(DoubleUnaryOperator function) {
    * dump the {@code <x,y>} pairs it computed found
    */
   public void dumpMemory() {
-    Double[] keys = memory.keySet().toArray(new Double[memory.keySet().size()]);
-    Arrays.sort(keys);
-    for (Double key : keys) {
-      log.info(key + "\t" + memory.get(key));
+    for (Map.Entry<Double, Double> entry : new TreeMap<>(memory).entrySet()) {
```
Comment (AngledLuffa) [inline src/edu/stanford/nlp/optimization/GoldenSectionLineSearch.java]: Here is another use of sorting via TreeMap.  It's just a debug method, but I don't like the functionality change

**Comment by AngledLuffa** [comment]
Alright, I merged this, but in general let me encourage you to spend your time on changes that make a more significant impact on the project
